pkg/kubelet/kubelet_node_status.go

```diff
@@ -602,9 +602,19 @@ func (kl *Kubelet) setNodeStatusMachineInfo(node *v1.Node) {
 		if initialCapacity != nil {
 			for k, v := range initialCapacity {
 				if v1helper.IsExtendedResourceName(k) {
+					glog.V(2).Infof("Update capacity for %s to %d", k, v.Value())
 					node.Status.Capacity[k] = v
 				}
 			}
+			// Remove stale extended resources.
+			for k := range node.Status.Capacity {
+				if v1helper.IsExtendedResourceName(k) {
+					if _, ok := initialCapacity[k]; !ok {
```

Comment:
nit: `initialCapacity` should be named as `currentCapacity`.

Reply:
done. PTAL.